### ISSUE TITLE
MD007 is applied to numbered lists

### ISSUE THREAD

**Issue by palto42** [rvben/rumdl#638]
### rumdl version

0.1.95

### Installation method

pip install

### Bug description

Since rumdl claims compatibility with markdownlint if `style = "strict"`, it shoudl align with the originla definition of [MD007](https://github.com/DavidAnson/markdownlint/blob/v0.40.0/doc/md007.md) and apply this rule only to unordered lists and not to numbered lists.

In numbered lists, the nested lists must be indented as least up to the first character of the text, which is at least 3 characters for single digit numbers.

If I configure MD007 with 2 characters, numbered lists will get linter errors.

### Steps to reproduce

1. Configre MD007 with 2 characters

2. Check this example, it will show an error for the indented text as it uses 3 characters.

```markdown
1. Some text
   - Indented text
     - more indented
``` 

3. Try strict indentation with 2 characters and check that it will not correctly render.

Example of list not correctly rendered

```markdown
1. Some text
  - Text not rendered as indented
    - more indented
```  

### Expected behavior

Only apply the rule for unordered lists and ignore numbered (ordered) lists.

### Actual behavior

Rule is applied to numbered lists, which is not aligned with original markdownlint rule.

### Configuration

```toml
[global]

flavor = "azure_devops"

enable = [
  "MD007", # Unordered list indentation
]

# MD007: Unordered list indentation
[MD007]
indent = 2
style = "fixed" # should be compatible with markdownlint
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in **v0.2.0**. MD007 now honors markdownlint's "all ancestors unordered" rule, so unordered sublists nested under an ordered list item are exempt at every depth.

Your example now reports clean:

```markdown
# Title

1. Some text
   - Indented text
     - more indented
```

`rumdl check` -> `No issues found` (with `[MD007] indent=2, style="fixed"`).

Thanks for the clear repro and the markdownlint-cli2 comparison. Closing as released in v0.2.0; reopen if you still see it.
